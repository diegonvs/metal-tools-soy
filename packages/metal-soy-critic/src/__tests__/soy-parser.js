const soyParser = require('../soy-parser');
const {getFixture} = require('../../test/util');

describe('Soy Parser', () => {
  test('should parse file correctly', () => {
    const input = getFixture('Test.soy');

    expect(soyParser(input)).toMatchSnapshot();
  });

  test('should fail to parse file', () => {
    const input = getFixture('Invalid.soy');

    expect(() => soyParser(input)).toThrow();
  });

  test('should parse nested calls', () => {
    const input = getFixture('NestedCalls.soy');

    expect(soyParser(input)).toMatchSnapshot();
  });

  test('should parse deltemplates', () => {
    const input = getFixture('DelTemplate.soy');

    expect(soyParser(input)).toMatchSnapshot();
  });

  test('should parse private templates', () => {
    const input = getFixture('PrivateTemplate.soy');

    expect(soyParser(input)).toMatchSnapshot();
  });
});
