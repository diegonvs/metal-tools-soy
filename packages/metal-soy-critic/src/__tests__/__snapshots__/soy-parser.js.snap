// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Soy Parser should parse file correctly 1`] = `
Object {
  "body": Array [
    Object {
      "body": Array [
        Object {
          "name": ".helper",
          "type": "Call",
        },
        Object {
          "name": "OtherTemplate.render",
          "type": "Call",
        },
      ],
      "name": ".render",
      "params": Array [
        Object {
          "name": "name",
          "paramType": "string",
          "required": true,
          "type": "Param",
        },
        Object {
          "name": "title",
          "paramType": "string",
          "required": true,
          "type": "Param",
        },
        Object {
          "name": "optionalInfo",
          "paramType": "string",
          "required": false,
          "type": "Param",
        },
      ],
      "type": "Template",
    },
    Object {
      "body": Array [],
      "name": ".helper",
      "params": Array [
        Object {
          "name": "test",
          "paramType": "asdasd",
          "required": false,
          "type": "Param",
        },
      ],
      "type": "Template",
    },
  ],
  "namespace": "Test",
  "type": "Program",
}
`;
